internal/services/shared/schema.go

```diff
@@ -318,9 +318,45 @@ func sanityCheckNamespaceChanges(
 	for _, delta := range diff.Deltas() {
 		switch delta.Type {
 		case nsdiff.RemovedRelation:
+			// NOTE: We add the subject filters here to ensure the reverse relationship index is used
+			// by the datastores. As there is no index that has {namespace, relation} directly, but there
+			// *is* an index that has {subject_namespace, subject_relation, namespace, relation}, we can
+			// force the datastore to use the reverse index by adding the subject filters.
+			var previousRelation *core.Relation
+			for _, relation := range existing.Relation {
+				if relation.Name == delta.RelationName {
+					previousRelation = relation
+					break
+				}
+			}
+
+			if previousRelation == nil {
+				return nil, spiceerrors.MustBugf("relation `%s` not found in existing namespace definition", delta.RelationName)
+			}
+
+			subjectSelectors := make([]datastore.SubjectsSelector, 0, len(previousRelation.TypeInformation.AllowedDirectRelations))
+			for _, allowedType := range previousRelation.TypeInformation.AllowedDirectRelations {
+				if allowedType.GetRelation() != "" {
```

Comment:
I think this worked by _happenstance_, but the ellipsis was not being handled correctly, as it's not modeled as empty string at this level of the stack, but `...`.

There is also no explicit handling of wildcards, but my understanding is that's modeled as an `userset_object_id = *`, but `userset_relation = ""`, so it ended up falling in ellipsis branch, which would generate the wrong `SubjectRelationFilter`.

I've force-pushed the changes, let me know what you think